cpp/src/aztec/ecc/curves/bn254/scalar_multiplication/pippenger.cpp

```diff
@@ -14,23 +13,27 @@ Pippenger::Pippenger(g1::affine_element* points, size_t num_points)
     scalar_multiplication::generate_pippenger_point_table(monomials_, monomials_, num_points);
 }
 
-Pippenger::Pippenger(uint8_t const* points, size_t num_points)
+Pippenger::Pippenger(uint8_t const* points, size_t num_points, bool is_lagrange)
     : num_points_(num_points)
 {
     monomials_ = point_table_alloc<g1::affine_element>(num_points);
 
-    monomials_[0] = barretenberg::g1::affine_one;
+    size_t index = 0;
+    if (!is_lagrange) {
```

Comment:
Note to self: this conditional logic to handle the generator $[1]_1 \in \mathbb{G}_1$ creates the need for `is_lagrange` variable. If we can add $[1]_1$ to the first monomial transcript, we can get rid of this `is_lagrange`. Do this as a fresh PR.